src/module_3/push_mvp.py

```diff
@@ -0,0 +1,375 @@
+
+
+# Initial column classification:
+predicted = ["outcome"]
+information = ["variant_id", "order_id", "user_id", "created_at", "order_date"]
+numerical = [
+    "user_order_seq",
+    "normalised_price",
+    "discount_pct",
+    "global_popularity",
+    "count_adults",
+    "count_children",
+    "count_babies",
+    "count_pets",
+    "people_ex_baby",
+    "days_since_purchase_variant_id",
+    "avg_days_to_buy_variant_id",
+    "std_days_to_buy_variant_id",
+    "days_since_purchase_product_type",
+    "avg_days_to_buy_product_type",
+    "std_days_to_buy_product_type",
+]
+
+categorical = ["product_type", "vendor"]
+binary = ["ordered_before", "abandoned_before", "active_snoozed", "set_as_regular"]
+
+# Define the functions
+
+
+def assess_NA(data: pd.DataFrame):
+    """
+    Returns a pd.DataFrame denoting the total number of NA
+    values and the percentage of NA values in each column.
+    """
+    # pd.Datadenoting features and the sum of their null values
+    nulls = data.isnull().sum().reset_index().rename(columns={0: "count"})
+    nulls["percent"] = nulls["count"] * 100 / len(data)
+
+    return nulls
+
+
+def read_data(file_path: str) -> pd.DataFrame:
+    """
+    Read and validate data from a CSV file.
+    Parameters: file_path (str)
+    Returns: pd.DataFrame
+    """
+    try:
+        data = pd.read_csv(file_path, index_col=False)
+
+        if data is not None and not data.empty:
+            logging.info("Data loaded successfully.")
+
+            # Info about its shape and nulls
+            rows, cols = data.shape
+            logging.info(f"Data shape: {rows} rows x {cols} columns")
+
+            null_assessment = assess_NA(data)
+            logging.info(
+                "Assessment of NA values:\n" + null_assessment.to_string(index=False)
+            )
+
+            # Show data sample
+            print(data.head(5))
+
+            return data
+        else:
+            logging.error("Error: The loaded data is empty or None")
+            return
+
+    except FileNotFoundError as e:
+        logging.error(f"Error: The CSV file is empty. Details: {e}")
+    except pd.errors.EmptyDataError as e:
+        logging.error(
+            f"Error: An unexpected error occurred while loading the data. Details {e}"
+        )
+
+
+def preprocess_data(
+    data: pd.DataFrame, remove_if_all_na: bool = False, num_items: int = 5
+) -> pd.DataFrame:
+    """
+    Preprocess data by removing rows with
+    filter orders with at least 5 items
+
+    Parameters:
+    - data (pd.DataFrame)
+    - remove_all_na_rows (bool): If True, remove rows where at least
+    1 value is missing. If False, remove rows where all values are missing.
+
+    Returns:
+    - pd.DataFrame. The preprocessed dataset
+    """
+    try:
+        initial_length = len(data)
+        if remove_if_all_na:  # remove if everything is NA
+            data = data.dropna(how="all")
+        else:
+            data = data.dropna()
```

Comment:
nitpick: for more clarity on the logic, you could add the `how` parameter (`any`) even if it's the default one